legend-engine-xts-relationalStore/legend-engine-xt-relationalStore-generation/legend-engine-xt-relationalStore-pure/src/main/resources/core_relational/relational/sqlQueryToString/extensionDefaults.pure

```diff
@@ -159,6 +159,13 @@ function meta::relational::functions::sqlQueryToString::default::convertGreatest
   )->joinStrings('case ', ' ', ' else null end'));
 }
 
+function meta::relational::functions::sqlQueryToString::default::convertBoolAndOrToCaseStatement(param:String[1], and:Boolean[1]):String[1]
+{
+  let fn = if ($and, | 'min', | 'max');
+
+  'case when ' + $fn + '(' + $param + ') = \'true\' then \'true\' else \'false\' end';
```

Comment:
hardcoding true/false as string seems wrong, and disregard the DB extension boolean support handling.

Reply:
valid concern which was considered. We do not actually have access to that flag at the required point. This function is designed to be more a shared helper which can be parameterized accordingly for the dialect if required. (E.g min/max, comparison etc). We need true/false return to ensure correct type handling on resultset streaming.

Reply:
1. that would introduce a completely asymmetric flow to any other dynafunc transform. There are many considerably more complex dynafunc transformw than this which are done in a similar manner.
2. you do not have all the context at the time to do that
3. this would add significant complexity for which is fundamentally a simple transform.

This function is a simple shared helper using common pattern to be used where needed for the dialects. It can either be modified if needed in future to handle special cases or simply coded directly in the specific dialect handler